### PR TITLE
Added C3 grammar

### PR DESCRIPTION
**Pull request by id3nom**

### CODE REVIEW

**Comment by id3nom** [comment]
> Can you add injections at least for comments?
> 
> And please add some description with links to the language, the parser, a Vim filetype, and some context.
> 
> (You probably want to add a README name as well since "c3" is not really descriptive.)

I will do so thanks for the feedback

**Comment by clason** [comment]
`tree-sitter generate` throws a warning:
> Warning: rule integer_literal is just a `seq` or `choice` rule with a single element. This is unnecessary.

Please address that; we've had issues with the generated parser not behaving correctly in this situation.

**Comment by clason** [comment]
And you should run `./scripts/format-queries.lua queries/c3` to pass the linter check.

**Comment by clason** [comment]
And you can only use the "standard captures" in listed https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#highlights.

File: README.md [hunk 1]
```diff
@@ -191,6 +191,7 @@ We are looking for maintainers to add more parsers and to write query files for
 - [x] [blueprint](https://gitlab.com/gabmus/tree-sitter-blueprint.git) (experimental, maintained by @gabmus)
 - [x] [bp](https://github.com/ambroisie/tree-sitter-bp) (maintained by @ambroisie)
 - [x] [c](https://github.com/tree-sitter/tree-sitter-c) (maintained by @amaanq)
+- [ ] [c3](https://github.com/c3lang/tree-sitter-c3) (maintained by @cbuttner)
```
Comment (ribru17) [inline README.md:194]: the maintainer here should be you as it refers to the queries maintainer, rather than parser maintainer

**Comment by clason** [comment]
@id3nom are you planning to make the required changes and finish this PR?

**Comment by clason** [comment]
Closing as stale; if you want to pick this up again, please ping me **before** pushing to your branch (otherwise the PR cannot be reopened).

**Pull request closed without merging**